src/components/VirtualList/Config.ts

```diff
@@ -0,0 +1,109 @@
+import deepFreeze from 'deep-freeze';
+import { AppListItem } from '../../types/AppList';
+import {
+  ApplicationsIcon,
+  AutomationIcon,
+  BundleIcon,
+  CodeBranchIcon,
+  GlobeRouteIcon,
+  IntegrationIcon,
+  LockedIcon,
+  MigrationIcon,
+  NetworkIcon,
+  ServiceIcon
+} from '@patternfly/react-icons';
+import { WorkloadListItem } from '../../types/Workload';
+import { ServiceListItem } from '../../types/ServiceList';
+import { IstioConfigItem } from '../../types/IstioConfigList';
+
+export type TResource = AppListItem | WorkloadListItem | ServiceListItem | IstioConfigItem;
+
+type resourceType = {
+  name: string;
+};
+
+const health: resourceType = {
+  name: 'Health'
+};
+
+const sidecar: resourceType = {
+  name: 'Sidecar'
+};
+
+const configuration: resourceType = {
+  name: 'Configuration'
+};
+
+const labelValidation: resourceType = {
+  name: 'LabelValidation'
+};
+
+const istioType: resourceType = {
+  name: 'IstioType'
+};
+
+const istioTypes = {
+  gateway: { name: 'Gateway', icon: GlobeRouteIcon },
+  virtualservice: { name: 'VirtualService', icon: CodeBranchIcon },
+  destinationrule: { name: 'DestinationRule', icon: NetworkIcon },
+  serviceentry: { name: 'ServiceEntry', icon: ServiceIcon },
+  rule: { name: 'Rule', icon: MigrationIcon },
+  adapter: { name: 'Adapter', icon: MigrationIcon },
+  template: { name: 'Template', icon: MigrationIcon },
+  quotaspec: { name: 'QuotaSpec', icon: AutomationIcon },
+  quotaspecbinding: { name: 'QuotaSpecBinding', icon: IntegrationIcon },
+  policy: { name: 'Policy', icon: LockedIcon },
+  meshpolicy: { name: 'MeshPolicy', icon: LockedIcon },
+  clusterrbacconfig: { name: 'ClusterRbacConfig', icon: LockedIcon },
+  rbacconfig: { name: 'RbacConfig', icon: LockedIcon },
+  sidecar: { name: 'Sidecar', icon: IntegrationIcon },
+  servicerole: { name: 'ServiceRole', icon: LockedIcon },
+  servicerolebinding: { name: 'ServiceRoleBinding', icon: LockedIcon }
+};
+
+export type resource = {
+  name: string;
+  columns?: string[];
+  information: resourceType[];
+  caption?: string;
+  istioTypes?: any;
```

Comment:
I'm not sure that `istioTypes` should be here in `resource` type, as it's not something that's going to change from a resource to another. What about removing it from resource, exporting the `const istioTypes` and directly consume this exported constant wherever it's used ?

Reply:
oh you're right